charts/consul/templates/telemetry-collector-v2-deployment.yaml

```diff
@@ -0,0 +1,379 @@
+{{- if and .Values.telemetryCollector.enabled (mustHas "resource-apis" .Values.global.experiments) }}
+{{- if not .Values.telemetryCollector.image}}{{ fail "telemetryCollector.image must be set to enable consul-telemetry-collector" }}{{ end }}
+{{- if not .Values.connectInject.enabled }}{{ fail "connectInject.enabled must be true" }}{{ end -}}
+{{- if and .Values.global.adminPartitions.enabled (not .Values.global.enableConsulNamespaces) }}{{ fail "global.enableConsulNamespaces must be true if global.adminPartitions.enabled=true" }}{{ end }}
+{{ template "consul.validateCloudSecretKeys" . }}
+{{ template "consul.validateTelemetryCollectorCloud" . }}
+{{ template "consul.validateTelemetryCollectorCloudSecretKeys" . }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "consul.fullname" . }}-telemetry-collector
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: consul-telemetry-collector
+    {{- if .Values.global.extraLabels }}
+      {{- toYaml .Values.global.extraLabels | nindent 4 }}
+    {{- end }}
+spec:
+  replicas: {{ .Values.telemetryCollector.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "consul.name" . }}
+      chart: {{ template "consul.chart" . }}
+      release: {{ .Release.Name }}
+      component: consul-telemetry-collector
+  template:
+    metadata:
+      annotations:
+        "consul.hashicorp.com/mesh-inject": "false"
+        # This annotation tells the pod controller that this pod was injected even though it wasn't. The
+        # endpoints controller would then sync the endpoint into Consul
+        "consul.hashicorp.com/mesh-inject-status": "injected"
+        # We aren't using tproxy and we don't have an original pod. This would be simpler if we made a path similar
+        # to gateways
+        "consul.hashicorp.com/transparent-proxy": "false"
+        "consul.hashicorp.com/transparent-proxy-overwrite-probes": "false"
+        "consul.hashicorp.com/connect-k8s-version": {{ $.Chart.Version }}
```

Comment:
AFAIK, we don't consume this anywhere, and it seems more like debugging metadata. I thought we should keep both in case external tooling was looking for this, but since I'm unaware of any and the pattern thus far has been to upgrade, I think we can drop `connect-k8s-version`.